src/pages/registration/files_and_license_tab/AdministrativeAgreementFilesRow.tsx

```diff
@@ -0,0 +1,82 @@
+import { Box, Checkbox, IconButton, TableCell, TableRow, Tooltip, Typography } from '@mui/material';
+import CancelIcon from '@mui/icons-material/Cancel';
+import { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import prettyBytes from 'pretty-bytes';
+import { Field, FieldProps, useFormikContext } from 'formik';
+import { AssociatedFile, AssociatedFileType } from '../../../types/associatedArtifact.types';
+import { ConfirmDialog } from '../../../components/ConfirmDialog';
+import { SpecificFileFieldNames } from '../../../types/publicationFieldNames';
+import { dataTestId } from '../../../utils/dataTestIds';
+import { TruncatableTypography } from '../../../components/TruncatableTypography';
+
+interface AdminstrativeAgreementFilesRowProps {
+  file: AssociatedFile;
+  removeFile: () => void;
+  toggleLicenseModal: () => void;
+  baseFieldName: string;
+}
+
+export const AdminstrativeAgreementFilesRow = ({
+  file,
+  removeFile,
+  baseFieldName,
+}: AdminstrativeAgreementFilesRowProps) => {
+  const { t } = useTranslation();
+  const [openConfirmDialog, setOpenConfirmDialog] = useState(false);
+  const toggleOpenConfirmDialog = () => setOpenConfirmDialog(!openConfirmDialog);
+  const { setFieldValue } = useFormikContext();
+
+  return (
+    <TableRow data-testid={dataTestId.registrationWizard.files.fileRow}>
+      <TableCell sx={{ minWidth: '13rem' }}>
+        <Box sx={{ display: 'flex', gap: '0.5rem', justifyContent: 'space-between', alignItems: 'center' }}>
+          <TruncatableTypography>{file.name}</TruncatableTypography>
+          <Tooltip title={t('registration.files_and_license.remove_file')}>
+            <IconButton onClick={toggleOpenConfirmDialog} sx={{ mr: '0.5rem' }}>
```

Comment:
Ikonet er lengst til høyre for navnet nå, så det skal vel ikke være noe margin til høyre her da det er dekt av gap på flexen?
```suggestion
            <IconButton onClick={toggleOpenConfirmDialog}>
```